rholang/src/test/scala/coop/rchain/rholang/interpreter/ErrorsSpec.scala

```diff
@@ -0,0 +1,26 @@
+package coop.rchain.rholang.interpreter
+import java.io.{ByteArrayOutputStream, PrintStream}
+
+import coop.rchain.rholang.interpreter.errors.{InterpreterError, UnrecognizedInterpreterError}
+import org.scalacheck.ScalacheckShapeless._
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+import org.scalatest.{Assertion, FlatSpec, Matchers}
+
+class ErrorsSpec extends FlatSpec with Matchers with GeneratorDrivenPropertyChecks {
+
+  "Using Throwable methods on InterpreterError" should "not cause exceptions itself" in {
+    forAll { e: InterpreterError =>
```

Comment:
So `shapeless` generated instances for all of the `InterpreterError` subclasses?

Reply:
Yes! :D

Reply:
One thing that got in the way when I was testing the test itself (by reverting the prod code changes) was the InterpreterError trait not being sealed. It works though for sealed traits, sealed abstract classes and - I bet - for case classes as long as their components have an `Arbitrary` instance.